recipe/wrapper.c

```diff
@@ -0,0 +1,261 @@
+
+// http://www.catch22.net/tuts/reducing-executable-size
+// Favour small code
+#pragma optimize("gsy", on)
+
+
+/*if you need it: here is a way to get specific path for different arch... */
+#if __SIZEOF_POINTER__ == 8
+# define ARCH "x64"
+#else
+# define ARCH "i386"
+#endif
+
+/*
+Change these to the *relative* paths your real executable is in
+In the end this is used as
+   <path env><RELATIVE_PATH><wrapper name><EXEC_EXTENSION>
+Slashes are need to be included here!
+<path env> really means one dir below the dir where the wrapper is. Which is
+true for <env>\Scripts and <env>\bin
+*/
+#define RELATIVE_PATH   "\\Library\\miktex\\texmfs\\install\\miktex\\bin\\"
+#define EXEC_EXTENSION  ".exe"
+#define DEBUG 0
+
+
+/* Only change soemthing below where the path is set in the end.. */
+
+int child_pid=0;
+
+
+int fail(char *format, char *data)
+{
+    /* Print error message to stderr and return 2 */
+    fprintf(stderr, format, data);
+    return 2;
+}
+
+
+char *quoted(char *data)
+{
+    int i, ln = strlen(data), nb;
+
+    /* We allocate twice as much space as needed to deal with worse-case
+       of having to escape everything. */
+    char *result = calloc(ln * 2 + 3, sizeof(char));
+    char *presult = result;
+
+    *presult++ = '"';
+    for (nb=0, i=0; i < ln; i++)
+    {
+        if (data[i] == '\\')
+            nb += 1;
+        else if (data[i] == '"')
+        {
+            for (; nb > 0; nb--)
+                *presult++ = '\\';
+            *presult++ = '\\';
+        }
+        else
+            nb = 0;
+        *presult++ = data[i];
+    }
+
+    for (; nb > 0; nb--)        /* Deal w trailing slashes */
+        *presult++ = '\\';
+
+    *presult++ = '"';
+    *presult++ = 0;
+    return result;
+}
+
+
+
+void pass_control_to_child(DWORD control_type)
+{
+    /*
+     * distribute-issue207
+     * passes the control event to child process (Python)
+     */
+    if (!child_pid) {
+        return;
+    }
+    GenerateConsoleCtrlEvent(child_pid, 0);
+}
+
+
+BOOL control_handler(DWORD control_type)
+{
+    /*
+     * distribute-issue207
+     * control event handler callback function
+     */
+    switch (control_type) {
+        case CTRL_C_EVENT:
+            pass_control_to_child(0);
+            break;
+    }
+    return TRUE;
+}
+
+
+int create_and_wait_for_subprocess(char* command) {
+    /*
+     * distribute-issue207
+     * launches child process (Python)
+     */
+    DWORD return_value = 0;
+    LPSTR commandline = command;
+    STARTUPINFOA s_info;
+    PROCESS_INFORMATION p_info;
+    ZeroMemory(&p_info, sizeof(p_info));
+    ZeroMemory(&s_info, sizeof(s_info));
+    s_info.cb = sizeof(STARTUPINFO);
+    // set-up control handler callback funciotn
+    SetConsoleCtrlHandler((PHANDLER_ROUTINE) control_handler, TRUE);
+    if (!CreateProcessA(NULL, commandline, NULL, NULL, TRUE, 0, NULL,
+                        NULL, &s_info, &p_info))
+    {
+        fprintf(stderr, "failed to create process.\n");
+        return 1;
```

Comment:
Yeah, that is definitely better. Returning 0 on success annoys me greatly :-)

If you were to submit this to conda-recipes/r-base I would be happy, otherwise I'll try to remember it next time I update R on Windows.